src/turnkeyml/files_api.py

```diff
@@ -400,17 +382,28 @@ def benchmark_files(
                     file=file_path_absolute,
                     build_model=not build_only,
                     model_type=build.ModelType.ONNX_FILE,
-                    unique_invocations={onnx_hash: invocation_info},
                     hash=onnx_hash,
                 )
 
+                # Add UniqueInvocationInfo
+                #  - is_target=True is required or else traceback wont be printed for
+                #      in the event of any errors
+                #  - execute=1 is required or else the ONNX model will be
+                #       skipped in later stages of evaluation
+                model_info.add_unique_invocation(
+                    invocation_hash=onnx_hash,
+                    is_target=True,
+                    input_shapes=input_shapes,
+                    executed=1,
+                )
+
                 # Begin evaluating the ONNX model
                 tracer_args.script_name = onnx_name
                 tracer_args.models_found[tracer_args.script_name] = model_info
                 explore_invocation(
                     model_inputs=onnx_inputs,
                     model_info=model_info,
-                    invocation_info=invocation_info,
+                    invocation_info=model_info.unique_invocations[onnx_hash],
```

Comment:
nit: the code could be a bit simpler if it was `ModelInfo.add_unique_invocation() -> UniqueInvocationInfo` (ie, the function returned the new UniqueInvocationInfo instance).

That way you wouldn't have to look it up in the dictionary a few lines of code later.